packages/ceramic-doctype-tile/src/tile-doctype.ts

```diff
@@ -114,6 +126,31 @@ export class TileDoctype extends Doctype {
         return TileDoctype._signDagJWS(record, did, doctype.owners[0])
     }
 
+    /**
+     * Calculates anchor nonce
+     */
+    private static async _calculateNonce(doctype: Doctype): Promise<number> {
+        if (doctype.context.ipfs == null) {
+            return 0 // return nonce 0 if IPFS is not available
+        }
+
+        let nonce = 0
+        let cid = doctype.head
+
+        while (cid != null) {
+            let record = (await doctype.context.ipfs.dag.get(cid)).value
+            if (DoctypeUtils.isAnchorRecord(record)) {
+                return nonce
+            }
+            if (DoctypeUtils.isSignedRecord(record)) {
+                record = (await doctype.context.ipfs.dag.get(record.link)).value
+            }
+            cid = record.prev
+            nonce++
+        }
+        return nonce
```

Comment:
This will make the nonce be higher with each new signed record that is added? That is not what I meant with the nonce at all and I'm not sure this actually solves the problem.

It should be something more simple like this:
```suggestion
        // If there hasn't been any update prior to this we should not set the nonce
        if (!doctype.state.next) return
        // Get the current nonce and increment it by one
        return (doctype.state.next.metadata?.nonce || 0)++
```
Assuming that we don't use the `anchor` property as mentioned above. This means that the nonce only takes affect on each record "height" and once a signed record has been anchored, the nonce is reset to 0.

Reply:
Thanks for the suggestions. I'll test this solution now. We should definitely specify the proposals like this in the future (e.g. somewhere in the story, etc).

Reply:
yeah, sorry I thought it was more clear that it was.